### PR TITLE
Unify material parameter update

### PR DESCRIPTION
**Pull request by reduz**
* Unifies how material parameters are updated.
* Single function, easier to maintain.
* Updates materials properly when textures change.

### CODE REVIEW

File: core/templates/rid_owner.h [hunk 1]
```diff
@@ -193,7 +193,7 @@ class RID_Alloc : public RID_AllocBase {
 			if (THREAD_SAFE) {
 				spin_lock.unlock();
 			}
-			if (validator_chunks[idx_chunk][idx_element] & 0x80000000) {
+			if ((validator_chunks[idx_chunk][idx_element] & 0x80000000) && validator_chunks[idx_chunk][idx_element] != 0xFFFFFFFF) {
```
Comment (reduz) [inline core/templates/rid_owner.h:196]: I also added this small fix, which corrects the error (said it was uninitialized when it was actually freed)

**Review by BastiaanOlij — APPROVED** [review]
Owh that cleaned up nicely :)

File: core/templates/rid_owner.h (hunk 1 above)
Comment (akien-mga) [reply core/templates/rid_owner.h:196]: FYI, there was this fix a week ago which also impacted this: https://github.com/godotengine/godot/commit/37db8d61845f1bf22d50939268aa8baa2663d06b#diff-aaa31181a1479fa3b274adc649b7bb6b3ac392c8bd2a623b39c3b88810b12051

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Calinou** [comment]
Does this PR supersede https://github.com/godotengine/godot/pull/50183?

**Pull request merged**